frappe/utils/redis_wrapper.py

```diff
@@ -238,22 +238,65 @@ def hget(self, name, key, generator=None, shared=False):
 			self.hset(name, key, value, shared=shared)
 		return value
 
-	def hdel(self, name, key, shared=False):
+	def hdel(
+		self,
+		name: str,
+		keys: str | list | tuple,
+		shared=False,
+		pipeline: redis.client.Pipeline | None = None,
+	):
+		"""
+		A wrapper around redis' HDEL command
+
+		:param name: The hash name
+		:param keys: the keys to delete
+		:param shared: shared frappe key or not
+		:param pipeline: A redis.client.Pipeline object, if this transaction is to be run in a pipeline
+		"""
 		_name = self.make_key(name, shared=shared)
 
-		if _name in frappe.local.cache:
-			if key in frappe.local.cache[_name]:
-				del frappe.local.cache[_name][key]
-		try:
-			super().hdel(_name, key)
-		except redis.exceptions.ConnectionError:
-			pass
+		if not isinstance(keys, (list, tuple)):
+			keys = (keys,)
+
+		name_in_local_cache = _name in frappe.local.cache
+
+		local_pipeline = False
+
+		if pipeline is None:
+			pipeline = self.pipeline()
+			local_pipeline = True
+
+		for key in keys:
+			if name_in_local_cache:
+				if key in frappe.local.cache[_name]:
+					del frappe.local.cache[_name][key]
+			try:
+				pipeline.hdel(_name, key)
+			except redis.exceptions.ConnectionError:
+				pass
+
+		if local_pipeline:
+			pipeline.execute()
+
+	def hdel_names(self, names: list | tuple, key: str):
+		"""
+		A function to call HDEL on multiple hash names with a common key, run in a single pipeline
+
+		:param names: The hash names
+		:param key: The common key
+		"""
+		pipeline = self.pipeline()
+		for name in names:
+			self.hdel(name, key, pipeline=pipeline)
+		pipeline.execute()
 
 	def hdel_keys(self, name_starts_with, key):
 		"""Delete hash names with wildcard `*` and key"""
+		pipeline = self.pipeline()
```

Comment:
also look at delete_keys/delete_values (used by frappe.clear_cache)

Write tests to ensure these changes with pipeline work. 

`test_caching.py` has (very few) tests for rediswrapper